### DISCUSSION TITLE
1.0.9.7

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#83] [Обновления]
- Переработана синхронизация громкости перевода с громкостью видео (теперь, значения не будут привязываться друг к другу в соотношение 1:1, а будут совместно меняться при изменение одного из значений, например: перевод 95% --> 94% = видео 20% --> 19%)
- Длина менюшки расширения увеличена до 350 пикселей
- Добавлена стандартная опция, выставляющая громкость оригинала на 15% (по умолчанию включена, стандартный процент громкости можно поменять в коде расширения. За это отвечает переменная "defaultVideoVolume")
- Добавлено "живое" обновление значения громкости перевода. т.е, теперь, если вы включите перевод, а потом измените громкость перевода и перезапустите перевод, то громкость будет та, которую вы только что установили (ранее, громкость была та, которая была установлена до загрузки страницы и менялась, только, после перезагрузки страницы)
- Сделано, чтобы громкость видео возвращалась после отключения перевода к той, что была до включения перевода (#82)
- Возвращено, чтобы слайдер громкости оригинала появлялся только с включенным переводом
- Исправлены некоторые косяки в синхронизации громкости перевода с громкостью видео в версии расширения "VOT Cloudflare"
- Повторяющиеся действия при выключение перевода вынесены в отдельную функцию
- Удален некоторый дублирующийся код

## What's Changed
* Исправление ошибок в vot-cli by @tyusha0 in https://github.com/ilyhalight/voice-over-translation/pull/80

## New Contributors
* @tyusha0 made their first contribution in https://github.com/ilyhalight/voice-over-translation/pull/80

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.0.9.6...1.0.9.7

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.0.9.7'>1.0.9.7</a>.</em>

**Comment by Gerodot** [comment]
Протестировал на Userscripts в Safari, ругается "Не удалось запросить перевод видео"

**Reply by ilyhalight** [reply]
> Протестировал на Userscripts в Safari, ругается "Не удалось запросить перевод видео"

Можешь скинуть, что пишет в консоли браузера, когда возникает ошибка и если не затруднит, попробуй 2-ю версию - VOT Cloudflare. Возможно там такой ошибки не будет. Сейчас нету возможности что-либо тестировать на макосе

**Reply by Gerodot** [reply]
Вывод в консоли в Safari (Версия 16.3 (18614.4.6.1.5), macOS Ventura 13.2):
<img width="1680" alt="Снимок экрана 2023-01-29 в 15 58 50" src="https://user-images.githubusercontent.com/1532819/215327773-9578844f-f926-4250-b29c-03f7be668e09.png">

**Reply by ilyhalight** [reply]
> Вывод в консоли в Safari (Версия 16.3 (18614.4.6.1.5), macOS Ventura 13.2): <img alt="Снимок экрана 2023-01-29 в 15 58 50" width="1680" src="https://user-images.githubusercontent.com/1532819/215327773-9578844f-f926-4250-b29c-03f7be668e09.png">

Странно, но не в твоем случае, не в случае с другим человеком нету какой-либо ошибки связанной с расширением в консоли. Мб в слудующем месяце руки дойдут и попробую себе накатить и там потихоньку разбираться

**Comment by Neovolter** [comment]
Было бы круто помимо дубляжа добавить автосабы с переводом . А то Я браузер использую лишь ради этих сабов в YT, а в нём как на зло видео постоянно лагают

**Comment by maxlan47** [comment]
Низкий поклон за ваши старания. Супер, что теперь с моей опере есть перевод видео)
Подскажите, а можно сделать пустую страницу. Что бы перевести локальный файл ?

**Reply by ilyhalight** [reply]
Нельзя т.к. у них есть проверка на определенные адреса сайтов, в число которых, локальный сайт не входит.

**Reply by ilyhalight** [reply]
> Нельзя т.к. у них есть проверка на определенные адреса сайтов, в число которых, локальный сайт не входит.

В таком случае сервер будет возвращать "Перевод не удался. Попробуйте позже" (примерно, такой текст. Точно, что, там, пишет - не помню)

Если брать обычный сайт, которого нету в списке, то будет возврашать "Не удалось запросить перевод видео"

**Reply by maxlan47** [reply]
Проще всего, получается в VK заливать для перевода. Очень Вам благодарен за создание и поддержку скрипта.

**Comment by vladcrash** [comment]
почините safari пожалуйста 🙏

**Reply by ilyhalight** [reply]
Обновитесь до одной из более новых версий, например до 1.4.0.3. Она должна работать с сафари

**Reply by vladcrash** [reply]
не помогло :(

**Reply by NullVerdict** [reply]
А можно лог консоли? Возможно удастся понять причину.
